data_diff/databases/base.py

```diff
@@ -430,10 +436,14 @@ def render_concat(self, c: Compiler, elem: Concat) -> str:
 
         # We coalesce because on some DBs (e.g. MySQL) concat('a', NULL) is NULL
         else:
-            items = [
-                f"coalesce({self.compile(c, Code(self.to_string(self.compile(c, expr))))}, '<null>')"
-                for expr in elem.exprs
-            ]
+            items = []
+            for expr in elem.exprs:
+                if isinstance(expr.type, Bytes):
+                    items.append(
+                        f"coalesce({self.compile(c, Code(self.normalize_bytes(self.compile(c, expr), Bytes)))}, '<null>')"
+                    )
+                else:
+                    items.append(f"coalesce({self.compile(c, Code(self.to_string(self.compile(c, expr))))}, '<null>')")
```

Comment:
Maybe we can use the normalize_value_by_type here so it automatically uses the correct normalize_xxx method for each type. Something like:
```suggestion
            items = [
                f"coalesce({self.compile(c, Code(self.normalize_value_by_type(self.compile(c, expr), expr.type)))}, '<null>')"
                for expr in elem.exprs
            ]
```

Reply:
Ah good point! I'll try that, thanks